phat/contracts/brick_profile_factory/lib.rs

```diff
@@ -76,16 +76,28 @@ mod brick_profile_factory {
             Ok(())
         }
 
-        /// Instantiate a user profile contract for caller. Only once for each account.
+        /// Import user profile addresses (only owner). This will overwrite existing addresses.
         #[ink(message)]
-        pub fn create_user_profile(&mut self) -> Result<()> {
-            let caller = self.env().caller();
+        pub fn import_user_profiles(
+            &mut self,
+            profiles: Vec<(AccountId, AccountId)>,
+        ) -> Result<()> {
+            self.ensure_owner()?;
+            self.users.extend(
+                profiles
+                    .into_iter()
+                    .map(|kv| (kv.0, BrickProfileRef::from_account_id(kv.1.into()))),
+            );
+            Ok(())
+        }
 
-            if self.users.contains_key(&caller) {
-                return Err(Error::NoDuplicatedUserProfile);
-            }
+        /// Instantiate a user profile contract for caller, overwrite the existing one.
+        /// Return old user profile contract if there is.
+        #[ink(message)]
+        pub fn force_create_user_profile(&mut self) -> Result<Option<AccountId>> {
```

Comment:
Can we import all EVM accounts and workflow from the previous profile?

Reply:
The previous profile has `dump_evm_account()` and `get_all_workflows()` to retrieve both accounts and workflows.

The `import_evm_account()` is disabled for security reasons, so users should generate new accounts (while they can get the tokens back from the old accounts since they got the keys);
You can just create new workflows with exactly the same args as before them it looks like "import".